lib/protocol-decoder.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+var proxyProtocol = require('proxy-protocol-js')
+
+function protocolDecoder (client, data) {
+  var proto = {}
+  // var buffer = Buffer.allocUnsafe(0)
+  // buffer = client.conn.read(null)
+  if (!data) return proto
+  // todo: checkProxiesList(client.conn, client.broker.trustedProxies)
+  var trustProxy = client.broker.trustProxy
+  var ipFamily
+  var conn = client.conn
+  var socket = conn.socket || conn
+  var headers = client.req && client.req.headers ? client.req.headers : null
+  if (trustProxy && headers) {
+    if (headers['x-real-ip']) proto.ipAddress = headers['x-real-ip']
+    else if (headers['x-forwarded-for']) proto.ipAddress = headers['x-forwarded-for']
+    client.connDetails.isWebsocket = true
+    client.connDetails.isProxied = true
+  } else if (trustProxy) {
+    var proxyProto
+    try {
+      proxyProto = proxyProtocol.V1BinaryProxyProtocol.parse(data)
+    } catch (V1BinaryProxyProtocolE) {
+      try {
+        proxyProto = proxyProtocol.V2ProxyProtocol.parse(data)
+      } catch (V2ProxyProtocolE) {
+        // empty;
```

Comment:
Why is this empty?

Reply:
V2ProxyProtocol.parse throws error in case of an invalid buffer and the "empty" is due to an habit with my lint rules, forbidding empty blocks.
But not in this project apparently, so this comment will be removed.

Reply:
I'm sorry, I don't understand. Will this error being thrown during normal operations?

Reply:
This behavior will slow down aedes by at least 50% - we need a solution that does not generate an exception. The best case would be to feed data into the proxy decoder until it can decode correctly, and then avoiding it completely.

Reply:
Allright, i admit it was the lazy approach ;)
Another one could be to check the packet signature and if valid aedes will pick the right parser, ( since the error is thrown by proxy-protocol.js library when the signature is invalid )